### PR TITLE
test: add retries to flaky sample tests

### LINKED ISSUE #1412: com.example.storage.bucket.DeleteBucketPubSubNotificationTest: testDeleteBucketPubSubNotification failed
This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/main/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: 57f9817917565ee1cb0dd56ff2ff143ceda061aa
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/4a4e2028-aa39-4569-96db-4270d8bcac8f), [Sponge](http://sponge2/4a4e2028-aa39-4569-96db-4270d8bcac8f)
status: failed
<details><summary>Test output</summary><br><pre>com.google.cloud.storage.StorageException: We encountered an internal error. Please try again.
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.translate(HttpStorageRpc.java:286)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.createNotification(HttpStorageRpc.java:1651)
	at com.google.cloud.storage.StorageImpl$4.call(StorageImpl.java:1421)
	at com.google.cloud.storage.StorageImpl$4.call(StorageImpl.java:1418)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:103)
	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
	at com.google.cloud.storage.StorageImpl.createNotification(StorageImpl.java:1417)
	at com.example.storage.bucket.DeleteBucketPubSubNotificationTest.testDeleteBucketPubSubNotification(DeleteBucketPubSubNotificationTest.java:87)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.google.cloud.testing.junit4.StdXCaptureRule$1.evaluate(StdXCaptureRule.java:78)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:364)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:237)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 503 Service Unavailable
POST https://storage.googleapis.com/storage/v1/b/gcloud-test-bucket-temp-5b039934-a523-4ce7-aef8-7f4b55f413c0/notificationConfigs
{
  "code" : 503,
  "errors" : [ {
    "domain" : "global",
    "message" : "We encountered an internal error. Please try again.",
    "reason" : "backendError"
  } ],
  "message" : "We encountered an internal error. Please try again."
}
	at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:146)
	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:118)
	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:37)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$1.interceptResponse(AbstractGoogleClientRequest.java:428)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1111)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:514)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:455)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:565)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.createNotification(HttpStorageRpc.java:1648)
	... 40 more
</pre></details>

### PR DESCRIPTION
**Pull request by frankyn**
Fixes #1412, #1580, #1502 ☕️

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,24 @@
...
+        <groupId>com.google.cloud</groupId>
+        <artifactId>google-cloud-core</artifactId>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>com.google.cloud</groupId>
+        <artifactId>google-cloud-core</artifactId>
+        <scope>test</scope>
+      </dependency>
+    </dependencies><!-- {x-version-update:google-cloud-storage:current} -->
```
Comment (BenWhitehead) [inline pom.xml:24]: We don't want this here, was it supposed to go into `samples/pom.xml`?

File: samples/snippets/src/test/java/com/example/storage/ITHmacSnippets.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ public void before() {
     cleanUpHmacKeys(ServiceAccount.of(HMAC_KEY_TEST_SERVICE_ACCOUNT));
   }
 
-  @Rule public MultipleAttemptsRule multipleAttemptsRule = new MultipleAttemptsRule(5);
+  @Rule public MultipleAttemptsRule multipleAttemptsRule = new MultipleAttemptsRule(5, 2000L);
```
Comment (BenWhitehead) [inline samples/snippets/src/test/java/com/example/storage/ITHmacSnippets.java:52]: Any specific reason why this needs a longer initial timeout compared to the others?

**Review by BenWhitehead — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (frankyn) [reply pom.xml:24]: whoops .... might have fat fingered an intellij button. Reverting.

File: samples/snippets/src/test/java/com/example/storage/ITHmacSnippets.java (hunk 2 above)
Comment (frankyn) [reply samples/snippets/src/test/java/com/example/storage/ITHmacSnippets.java:52]: Failed consistently 5 times (https://source.cloud.google.com/results/invocations/1b7ca05f-7f48-44d3-9a99-ebeb4fda0cec/targets/github%2Fjava-storage%2Fsamples%2Fsnapshot%2Ftarget%2Ffailsafe-reports%2Fsamples%2Fcom.example.storage.ITHmacSnippets/tests); considered increasing the initial retry to give the API time on the first attempt only;

IIUC, @ Before will be called before each attempt and clean up resources. The issue with this multiple executions of integration tests and this clean up step doesn't consider age of HMAC resources created.

Comment (BenWhitehead) [reply samples/snippets/src/test/java/com/example/storage/ITHmacSnippets.java:52]: Fair enough. Yes, `@Before` will be called each attempt.

**Review by BenWhitehead — APPROVED** [review]

**Pull request merged**